### ISSUE TITLE
[REQ] [Kotlin][Client] Option to remove Response<> wrapper in operation return types

### ISSUE THREAD

**Issue by WonderCsabo** [OpenAPITools/openapi-generator#15491]
Currently if you use Coroutines (`suspend` modifier), your generated API functions will look like this:
```kotlin
@GET("users/{id}")
suspend fun user(@Path("id") id: Long): Response<User>
```

However there is another way to define the return type of the endpoint. You can omit the `Response<>` wrapper, and just return the `User`, like this:

```kotlin
@GET("users/{id}")
suspend fun user(@Path("id") id: Long): User
```

This is described [here](https://github.com/square/retrofit/blob/master/CHANGELOG.md#version-260-2019-06-05) in the Retrofit documentation.

It would be great if OpenAPI generator offer this as well. It could be controlled by a CLI option, and it could generated the current `Response<User>` form by default.

**Comment by WonderCsabo** [comment]
@jimschubert @dr4ke616 @karismann @Zomzog @andrewemery @4brunu @yutaka0m

In case you accept this feature, i would gladly contribute this.

**Comment by 4brunu** [comment]
Hi, thanks for taking the time to improve openapi-generator.
We try to avoid creating CLI options, because when we have lots of them, it's hard to maintain all the possible configurations.
I'm not sure, I think we could add this option to have direct access to the response, although that way we lose some information from the response.

**Comment by WonderCsabo** [comment]
I understand the code which is generated is opinionated. But it should be consistent (currently Rx and coroutines are different on this matter). 

For my use case, i would prefer without the wrapper of course. But i see there are other users of this tool.

**Comment by WonderCsabo** [comment]
Yes, that would work for me.  BTW, this option would only apply to the `retrofit2` library.

**Comment by 4brunu** [comment]
It would be nice to apply to all configurations, but we can expand it in the future

**Comment by WonderCsabo** [comment]
To be honest i do not even know if this is applicable to the other libraries.

**Comment by 4brunu** [comment]
Me neither because I'm not in front of a pc.
Can you please open a PR with it?

**Comment by lsuski** [comment]
Any update on that? Removing `Response` should be rather applicable only to `retrofit2` library because `Response` is Retrofit specific class. However it could be also aplicable to `useRxJava3` option. We could add an option `useResponseAsApiReturnType` or sth like that
